### ISSUE TITLE
"Unsaved Changes" is always visible, even after saving

### ISSUE THREAD

**Issue by seveibar** [tscircuit/snippets#326]
![Image](https://github.com/user-attachments/assets/9467f899-0865-478e-a1e8-18040c8c3020)

/bounty $10

**Comment by RohittCodes** [comment]
@seveibar @imrishabh18 not sure but maybe an issue in production. works fine locally
https://github.com/user-attachments/assets/fe38d4ef-5001-444c-84be-11304aea4baa

**Comment by krushnarout** [comment]
> [@seveibar](https://github.com/seveibar) [@imrishabh18](https://github.com/imrishabh18) not sure but maybe an issue in production. works fine locally https://github.com/user-attachments/assets/fe38d4ef-5001-444c-84be-11304aea4baa

Correct! But the UX is bad after clicking on save button it still show unsaved changes for a while.

But I fixed it #328 

https://github.com/user-attachments/assets/a8005c2d-1752-4613-aa45-58308fad1fa5

**Comment by seveibar** [comment]
It sounds like we need to validate that this works using the vercel preview url, if the issue still exists on production then the issue isnt resolved

**Comment by divanshu-go** [comment]
@seveibar I think we are not able to login on vercel preview url i have checked it see video for reference:
https://github.com/user-attachments/assets/3c6cb27f-ee37-4220-b223-35eddb660752

**Comment by ArnavK-09** [comment]
any updates on this issue? is it still valid? that tests?

**Comment by seveibar** [comment]
@ArnavK-09 thanks for following up on this but this issue had been fixed some time ago
